compiler/plugins/target/AMD-AIE/aie/AMDAIEObjectFifoStatefulTransform.cpp

```diff
@@ -330,7 +330,19 @@ SmallVector<BufferOp> createBuffers(OpBuilder &builder,
   SmallVector<BufferOp> buffers;
   if (deviceModel.isShimTile(tile.getCol(), tile.getRow())) return buffers;
   auto fifoType = cast<AIEObjectFifoType>(createOp.getElemType());
-  auto elemType = cast<MemRefType>(fifoType.getElementType());
+  MemRefType fifoElemType = cast<MemRefType>(fifoType.getElementType());
+  ArrayRef<int64_t> shape = fifoElemType.getShape();
+  Type elType = fifoElemType.getElementType();
+
+  Attribute memSpace = [&]() -> Attribute {
+    if (!fifoElemType.getMemorySpace()) return Attribute{};
+    // row 0: memspace 0,
+    // row 1: memspace 1,
+    // else : memspace 2.
+    return builder.getI32IntegerAttr(std::min<int>(2, tile.getRow()));
+  }();
```

Comment:
just curious: why do it like this? `Attributes are reference types already so you can just do 

```cpp
Attribute memSpace;
if (fifoElemType.getMemorySpace())
  memSpace = builder.getI32IntegerAttr(std::min<int>(2, tile.getRow()));
```

and save yourself the stack allocation for the lambda...

Also shouldn't it be

```cpp
  Attribute memSpace;
  if (fifoElemType.getMemorySpace())
    memSpace = fifoElemType.getMemorySpace();
```

instead of hard-coded `2`?

Reply:
I could. No reason I've done it this way other than it started as a 10-liner lambda until I whittled it down to this. And my mental stack for tracking uninitialized variables across multiple lines is especially low...

> and save yourself the stack allocation for the lambda...
I hope 'the compiler' in lines this, but tbh I'm not sure

Reply:
i mean it does inline - you get an inlined struct allocation for the lambda. it's not big but it's probably the size of `std::funciton` or something.

Reply:
https://en.cppreference.com/w/cpp/language/lambda

> The lambda expression is a prvalue expression of unique unnamed non-union non-aggregate class type, known as closure type, which is declared (for the purposes of ADL) in the smallest block scope, class scope, or namespace scope that contains the lambda expression.